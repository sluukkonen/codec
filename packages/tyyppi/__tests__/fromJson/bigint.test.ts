import { fromJson } from "../../src/index.js"
import { expectParseFailure, expectParseSuccess } from "../helpers.js"

test("should parse valid strings", () => {
  expectParseSuccess(fromJson.bigint, "0", 0n)
  expectParseSuccess(fromJson.bigint, "123", 123n)
  expectParseSuccess(fromJson.bigint, "-1000", -1000n)
  expectParseSuccess(fromJson.bigint, "+1000", 1000n)
})

test("should not accept leading or trailing whitespace", () => {
  expectParseFailure(fromJson.bigint, "  123")
  expectParseFailure(fromJson.bigint, "123   ")
  expectParseFailure(fromJson.bigint, "   123   ")
})

test("should not accept binary, octal and hexadecimal numbers", () => {
  expectParseFailure(fromJson.bigint, "0b11")
  expectParseFailure(fromJson.bigint, "0o53")
  expectParseFailure(fromJson.bigint, "0xFF")
  expectParseFailure(fromJson.bigint, "0xff")
})

test("should only accept integers", () => {
  expectParseFailure(fromJson.bigint, "123.45")
})

test("should not parse invalid strings", () => {
  expectParseFailure(fromJson.bigint, "")
  expectParseFailure(fromJson.bigint, "   ")
  expectParseFailure(fromJson.bigint, "asdf")
})

test("should not parse non-strings", () => {
  expectParseFailure(fromJson.bigint, null)
})
