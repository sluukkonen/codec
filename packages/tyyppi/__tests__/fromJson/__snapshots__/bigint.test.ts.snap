// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should ignore leading and trailing whitespace 1`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should ignore leading and trailing whitespace 2`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should ignore leading and trailing whitespace 3`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should not accept binary, octal and hexadecimal numbers 1`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should not accept binary, octal and hexadecimal numbers 2`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should not accept binary, octal and hexadecimal numbers 3`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should not accept binary, octal and hexadecimal numbers 4`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should not accept leading or trailing whitespace 1`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should not accept leading or trailing whitespace 2`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should not accept leading or trailing whitespace 3`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should not parse invalid strings 1`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should not parse invalid strings 2`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should not parse invalid strings 3`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;

exports[`should not parse non-strings 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "message": "Expected a string, received null",
    "path": [],
    "received": "null",
  },
]
`;

exports[`should only accept integers 1`] = `
[
  {
    "code": "invalid_integer_string",
    "message": "Expected the string to be an integer",
    "path": [],
  },
]
`;
